### PR TITLE
Improve type visibility of DCR object components

### PR DESCRIPTION
**Pull request by shtukas**
## What does this change?

Doesn't add or remove any behaviour. Just add some type annotations to help with understanding what's in the DCR data object and how it's put together.

### CODE REVIEW

File: common/app/model/dotcomrendering/DotcomRenderingUtils.scala [hunk 1]
```diff
@@ -295,9 +295,11 @@ object DotcomRenderingUtils {
       pageType: PageType,
   ): DotcomRenderingDataModel = {
 
-    val article = page.article
+    // Note: To help with readability and discoverability, all the values in this function have their types made explicit.
```
Comment (buck06191) [inline common/app/model/dotcomrendering/DotcomRenderingUtils.scala:298]: Excellent

File: common/app/model/dotcomrendering/DotcomRenderingUtils.scala [hunk 2]
```diff
@@ -476,74 +478,72 @@ object DotcomRenderingUtils {
...
+      sectionName = article.metadata.section.map(_.value), // Option[String]
+      subMetaSectionLinks = article.content.submetaLinks.sectionLabels.map(SubMetaLink.apply), // List[SubMetaLink]
+      subMetaKeywordLinks = article.content.submetaLinks.keywords.map(SubMetaLink.apply), // List[SubMetaLink]
+      shouldHideAds = article.content.shouldHideAdverts, // Boolean
+      isAdFreeUser = views.support.Commercial.isAdFree(request), // Boolean
+      webURL = article.metadata.webUrl, // String
+      linkedData = linkedData, // List[LinkedData]
+      openGraphData = openGraphData, // Map[String String]
+      twitterData = twitterData, // Map[String, String]
+      config = combinedConfig, // JsObject
```
Comment (buck06191) [inline common/app/model/dotcomrendering/DotcomRenderingUtils.scala:525]: Note that this one is the one we want to enter into and consider typing its fields explicitly in a future PR

**Review by buck06191 — APPROVED** [review]
Great work. Much better now that it's explicit about this

File: common/app/model/dotcomrendering/DotcomRenderingUtils.scala (hunk 2 above)
Comment (shtukas) [reply common/app/model/dotcomrendering/DotcomRenderingUtils.scala:525]: Understood, yeah, that's a nice follow up :)

**Pull request merged**